From b287a338dcc2e411098a421d33c9e92019578ece Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Tue, 22 Nov 2022 10:56:08 +0100
Subject: Avoid leaking by using Str function

strdup allocates and needs a free() call. Replace it with a Str function
which uses garbage collection in the background.
---
 etc.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/etc.c
+++ b/etc.c
@@ -2066,7 +2066,7 @@ FQDN(char *host)
 	for (res = res0; res != NULL; res = res->ai_next) {
 	    if (res->ai_canonname) {
 		/* found */
-		namebuf = strdup(res->ai_canonname);
+		namebuf = Strnew_charp(res->ai_canonname)->ptr;
 		freeaddrinfo(res0);
 		return namebuf;
 	    }
